app/support/message_users.rb

```diff
@@ -1,61 +1,34 @@
 # frozen_string_literal: true
 class MessageUsers
-  def self.call(type:, users:, template:, subject: nil)
-    new(type, users, template, subject).call
+  def self.call(type:, users:, template:, subject: nil, data: {})
+    new(type, users, template, subject, data).call
   end
 
-  def initialize(type, users, template, subject)
+  def initialize(type, users, template, subject, data)
     @type = type
     @users = users
     @template = template
     @subject = subject
+    @data = data
   end
 
   def call
     @users.each do |user|
-      attributes = user.attributes.symbolize_keys
-      begin
-        message = @template % attributes
-        subject = @subject % attributes
-      rescue KeyError => e
-        return { success: false, message: "Unknown key: '#{e.message}'" }
+      I18n.with_locale(user.locale) do
```

Comment:
`I18n.with_locale` doesn't work for `BaseNotifier` thats why there is a `BaseNotifier#with_locale` method (which is unfortunately not thread safe 👎). We should double check that this works..